### PR TITLE
[display] list @:deprecated meta in display/completion responses

### LINKED ISSUE #8178: [display] @:deprecated typedefs in completion
It's rather annoying to have the now deprecated types (#7390) show up in completion, and _before_ the ones you're meant to use:

I think there's no reason to show `@:deprecated` typedefs in completion at all? At least as long as they're just simple aliases, typedefs to anon structs should probably still be shown. I think there's already some detection for the difference for nicer auto-generated deprecation messages.

### PR DESCRIPTION
**Pull request by RealyUniqueName**
Closes #8178 
Also, removes "pos" field from "meta" structures.

Edit:
Changed this PR to allows an optional "meta" field in the "params" field of request json:
```json
{
	"jsonrpc": "2.0",
	"id": 13,
	"method": "display/completion",
	"params": {
		"meta": [
			":deprecated",
			":whatever"
		],
```

### CODE REVIEW

File: src/core/json/genjson.ml [hunk 1]
```diff
@@ -159,6 +158,12 @@ and generate_metadata ctx ml =
 	) ml in
 	jlist (generate_metadata_entry ctx) ml
 
+and generate_minimum_metadata ctx ml =
+	let ml = List.filter (fun (m,_,_) ->
+		m = Meta.Deprecated
```
Comment (Gama11) [inline src/core/json/genjson.ml]: Not sure how I feel about hardcoding it, maybe a filter array could be added to the request params so the client can choose what it wants to receive.

**Comment by Simn** [comment]
I don't know if removing `pos` is the right thing. It seems quite arbitrary to exclude that and I don't understand the reason for doing so.

**Comment by Gama11** [comment]
Well, I'd imagine it reduces the size quite a bit depending on how many metas are sent. And I still can't think of a good use case for including it.

**Comment by Simn** [comment]
There could be some "Remove metadata" code action or something, I don't know. And since when do we worry about the size of this stuff?

**Comment by Gama11** [comment]
We've always worried about that, otherwise there wouldn't have been any reason to implement resolve requests. :)

Completion requests aren't really useful for code actions, that's something for diagnostics.

**Comment by Simn** [comment]
Yes but this PR changes how metadata is generated for everything.

**Comment by Gama11** [comment]
I see. Yeah, I just wanted to omit it from completion results.

**Comment by RealyUniqueName** [comment]
I've changed this PR.
Now it allows optional "meta" field in the "params" field of request json:
```json
{
	"jsonrpc": "2.0",
	"id": 13,
	"method": "display/completion",
	"params": {
		"meta": [
			":deprecated",
			":whatever"
		],
```

**Comment by Gama11** [comment]
I'm wondering what to do in case we ever want to request all metas / no filtering.. Maybe it's fine to use `null` / "missing" for that? That would change the current default behavior, but only vshaxe implements the protocol atm anyway and we already default to "as verbose as possible" if `supportsResolve` isn't specified..

File: src/core/json/jsonrpc_handler.ml [hunk 2]
```diff
@@ -83,4 +84,17 @@ class jsonrpc_handler (id,name,params) = object(self)
 		try f() with JsonRpc_error _ -> def
 
 	method get_params = params
+
+	method get_requested_meta_list : string list =
```
Comment (Gama11) [inline src/core/json/jsonrpc_handler.ml]: Does this really belong in here? I think this is for general JSON-RPC handling, so also eval debugging?

Comment (RealyUniqueName) [reply src/core/json/jsonrpc_handler.ml]: Moved it to a separate class

**Comment by Gama11** [comment]
Without any `meta` filter set, this branch sends `meta: []` in responses, which seems misleading. Previously, the `meta` field was omitted entirely.

If a filter is set, `pos` is still included, but I guess it could be a future enhancement / optimization to omit it, so it's fine for now. It's probably not that bad if only the metas the client requests are sent.

I've updated the language server for this.

**Comment by RealyUniqueName** [comment]
> Without any meta filter set, this branch sends meta: []

Not anymore

**Pull request merged**